# Day 10 & 11 & 12 RNA sequencing of LINE 1 elements using GEO L1TD1 samples

- [x]  FeatureCounts values normalization
- [x]  Normal, Adenoma, Primary Graph --> L1TD1 gene expression

## FeatureCounts value normalization

- [TPM, FPKM, or Normalized Counts](https://translational-medicine.biomedcentral.com/articles/10.1186/s12967-021-02936-w)
- [RNA seq analysis](https://sbc.shef.ac.uk/workshops/2019-01-14-rna-seq-r/rna-seq-preprocessing.nb.html) in R
    - TPM R packages
    - [FPKM in R packages](https://cran.r-project.org/web/packages/countToFPKM/countToFPKM.pdf)
- Gene expression analysis using [high-throughput sequencing](https://compgenomr.github.io/book/gene-expression-analysis-using-high-throughput-sequencing-technologies.html#quantification) technologies
- methods in [edgeR](http://www.statsci.org/smyth/pubs/QLedgeRPreprint.pdf)

[CPM](https://www.rdocumentation.org/packages/edgeR/versions/3.14.0/topics/cpm) and [TMM](https://www.biostars.org/p/317701/) Normalization method

```r
library(edgeR)
dgList <- calcNormFactors(fc_result, method="TMM")
tmm <- cpm(fc_result,dgList, log=TRUE, prior.count=5)
```

R script used for loading data table, normalization, plotting

```r
fc_result_list <- read.table("/storage2/jwlab/yerim/rnaseq/fc_output/fc_result_list", quote="\"", comment.char="")
fc_result_list$name <- unlist(sapply(strsplit(fc_result_list$V1, "/", fixed=TRUE), function(x) x[7], simplify=FALSE))
SraRunTable <- read.csv ("/storage2/jwlab/yerim/rnaseq/fc_output/SraRunTable.txt" )
fc_result_list$source <- unlist(SraRunTable[match(fc_result_list$name,SraRunTable$Run),"Tissue"])
list <- fc_result_list[,-1]
rownames(list) <- list[,1]
listwithout190 <- list[c(-1),]

SRR191_fc_result <- read.delim("/storage2/jwlab/yerim/rnaseq/fc_output/SRR13401191/fc_result.txt", comment.char="#")
result <- SRR191_fc_result[,c(1,7)]
colnames(result)[2] <- "SRR13401191"
result <- cbind(result, SRR192_fc_result$X.storage2.jwlab.yerim.rnaseq.BAM.SRR13401192.Aligned.sortedByCoord.out.bam)
colnames(result)[3] <- "SRR13401192"
rownames(result) <- result$Geneid
result <- result[,c(-1)]

dgList <- calcNormFactors(result, method="TMM")
tmm <- cpm(result,dgList, log=TRUE, prior.count=5)

tmm <- cbind(tmm, SRR191_fc_result$Geneid)
colnames(tmm)[15] <- "Geneid"
grep("ENSG00000240563", row.names(tmm), value = TRUE)
# ENSG00000240563.2
L1TD1 <- tmm[row.names(tmm) == "ENSG00000240563.2"]

L1TD1 <- cbind(L1TD1, listwithout190$name)
colnames(L1TD1)[2] <- "name"
L1TD1 <- cbind(L1TD1, listwithout190$source)
colnames(L1TD1)[3] <- "sample"

library(ggplot2)
vplot <- ggplot(L1TD1, aes(x=sample, y=value, group=sample, fill=sample)) 
vplot <- vplot + geom_violin(trim=FALSE, fill="gray") 
vplot <- vplot + labs(title="L1TD1 expression in normal colon epithelium vs adenoma vs primary CRC", x="Sample Identities", y = "log2(CPM+1)") 
vplot <- vplot + geom_boxplot(width=0.1)+ theme_classic() + scale_x_discrete(limits=c("normal colon epithelium tissue", "adenoma tissue", "primary CRC tissue"))
```

![L1TD1](https://github.com/yerimeeei/Bioinformatics/assets/134043926/9b66ab6f-f658-44a0-b276-ddeb9f02b085)

- ****[SSHFS: How to Mount Remote File Systems Over SSH](https://phoenixnap.com/kb/sshfs)****
    - bring file from ssh server without downloading
- [Samtools flagstat](https://davetang.org/wiki/tiki-index.php?page=SAMTools#Simple_stats_using_SAMTools_flagstat)

### Differential Analysis

1. featureCounts run SRR13401190 (patient 5) except -p
2. Patients 1-4 뽑고
3. Differential Analysis [https://lashlock.github.io/compbio/R_presentation.html](https://lashlock.github.io/compbio/R_presentation.html)
4. abs(log2foldchange) > 2 and p.adj < 0.05 --> red coloredp.adj < 0.05 --> blue colored

[https://rstudio-pubs-static.s3.amazonaws.com/329027_593046fb6d7a427da6b2c538caf601e1.html](https://rstudio-pubs-static.s3.amazonaws.com/329027_593046fb6d7a427da6b2c538caf601e1.html)

5. Check if L1TD1 exists

6. If multiple conditions are not working, normal vs tumor, normal vs adenoma, adenoma vs tumor

- R Codes

```r
p1_4_result <- result[,c(-5,-10)]
p1_4_list <- list[c(-1, -6, -11),]

# if (!require("BiocManager", quietly = TRUE))
#   install.packages("BiocManager")
#
# BiocManager::install("DESeq2")
# DESeq2 Differential Analysis
library(DESeq2)
all(colnames(p1_4_result) %in% rownames(p1_4_list))
all(colnames(p1_4_result) == rownames(p1_4_list))
dds <- DESeqDataSetFromMatrix(countData = p1_4_result, colData = p1_4_list, design = ~ source)
# To remove low counts (less than 10)
# keep <- rowSums(counts(dds)) >= 10
# dds2 <- dds[keep,]
dds$source <- relevel(dds$source, ref = "normal colon epithelium tissue")
dds <- DESeq(dds)
res <- results(dds)
summary (res)
head(results(dds, tidy=TRUE))
res <- res[order(res$padj),]
head(res)
results <- results(dds, contrast = c("source", "normal colon epithelium tissue", "primary CRC tissue"))
res1 <- results(dds, contrast = c("source", "normal colon epithelium tissue", "primary CRC tissue"))
res2 <- results(dds, contrast = c("source", "normal colon epithelium tissue", "adenoma tissue"))
res3 <- results(dds, contrast = c("source", "primary CRC tissue", "adenoma tissue"))

#plotCounts
plotCounts <- plotCounts(dds, gene="ENSG00000240563.2", intgroup="source")

#Volcano plots
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-3,3)))
with(subset(res, padj<.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(res, padj<.05 & abs(log2FoldChange)>2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))

volp_NvsP <- with(res1, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-3,3)))
volp_NvsP <- with(subset(res1, padj<.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
volp_NvsP <- with(subset(res1, padj<.05 & abs(log2FoldChange)>2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))

volp_NvsA <- with(res2, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-3,3)))
volp_NvsA <- with(subset(res2, padj<.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
volp_NvsA <- with(subset(res2, padj<.05 & abs(log2FoldChange)>2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))

volp_PvsA <- with(res3, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-3,3)))
volp_PvsA <- with(subset(res3, padj<.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
volp_PvsA <- with(subset(res3, padj<.05 & abs(log2FoldChange)>2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))

# detect for L1TD1
# library(org.Hs.eg.db)
# res.df <- as.data.frame(res1)
# res.df$symbol <- mapIds(org.Hs.eg.db, keys = rownames(res.df), keytype = "ENSEMBL", column = "SYMBOL")
# -> error
selected = c("ENSG00000240563.2")
vol1 <- EnhancedVolcano(res1, x = "log2FoldChange", y = "padj", lab = row.names(res1), selectLab = selected, title = 'Volcano Plot of Normal Colon Epithelium Tissue vs Primary CRC Tissue')
vol2 <- EnhancedVolcano(res2, x = "log2FoldChange", y = "padj", lab = row.names(res2), selectLab = selected, title = 'Volcano Plot of Normal Colon Epithelium Tissue vs Adenoma Tissue')
vol3 <- EnhancedVolcano(res3, x = "log2FoldChange", y = "padj", lab = row.names(res3), selectLab = selected, title = 'Primary CRC Tissue vs Adenoma Tissue')

# Shrink gene-wise dispersion estimates toward the values predicted by the curve
plotDispEsts(dds)

dds <- makeExampleDESeqDataSet(n=100,m=6)
dds$source <- factor( c( "normal colon epithelium tissue", "primary CRC tissue", "adenoma tissue") )
dds <- DESeq(dds)
res = results(dds, contrast=c("source","normal colon epithelium tissue", "primary CRC tissue"))
res <- res[order(res$padj),]
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-3,3)))
with(subset(res, padj<.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(res, padj<.05 & abs(log2FoldChange)>2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
```

![N_vs_A](https://github.com/yerimeeei/Bioinformatics/assets/134043926/3eeb8d09-e1d4-4114-a94f-a51a2b02a0dd)

![N_vs_P](https://github.com/yerimeeei/Bioinformatics/assets/134043926/970d3b7a-dcf9-427b-b64e-6d236e9402fe)

![P_vs_A](https://github.com/yerimeeei/Bioinformatics/assets/134043926/2cd675af-8202-4c68-9c8f-2bb6d270fdd5)
