# Day 18 & 19 & 20 L1TD1 Gene Expression Single Cell Analysis

- [x]  **paired sample (N vs A) differential analysis**
- [x]  **venn diagram for (N vs A) & (A vs CA)**
- [x]  **motif analysis of upstream +/-**
- [x]  motif analysis TF affinity score distribution graph

### motifs in L1TD1 promoter (TSS +/-)

| name | class | family |  |
| --- | --- | --- | --- |
| ETV5::FOXI1 | Tryptophan cluster factors::Fork head/winged helix factors | Ets-related::FOX | 10.8 |
| HOXC11 | Homeo domain factors | HOX | 10.1 |
| HMBOX1 | Homeo domain factors | POU domain factors | 9.42 |
| IKZF1 | C2H2 zinc finger factors | Factors with multiple dispersed zinc fingers | 9.32 |
| HOXC9 | Homeo domain factors | HOX | 9.27 |
| POU2F1 | Homeo domain factors | POU domain factors | 8.89 |
| ZNF331 | C2H2 zinc finger factors | More than 3 adjacent zinc fingers | 8.80 |
| ERF::FOXO1 | Tryptophan cluster factors::Fork head/winged helix factors | Ets-related::FOX | 8.65 |
| GCM2 | GCM domain factors  | GCM | 8.48 |
| NR2F1 | Nuclear receptors with C4 zinc fingers  | RXR-related receptors (NR2)  | 8.44 |

### **paired sample (N vs A & A vs P) differential analysis**

```r
p1_4_list <- cbind(p1_4_list, SraRunTable$subject_status.id[c(-1,-6,-11)])
colnames(p1_4_list)[3] <- "patient.id"
# patient.id -> paired sample
dds <- DESeqDataSetFromMatrix(countData = result_14, colData = p1_4_list, design = ~ patient.id + source)
dds$source <- relevel(dds$source, ref = "adenoma tissue")
dds <- DESeq(dds)
# padj value < 0.05
respa <- results(dds, contrast = c("source", "primary CRC tissue", "adenoma tissue", alpha = 0.05))
resnp <- results(dds, contrast = c("source", "normal colon epithelium tissue", "adenoma tissue"), alpha = 0.05)

selected = c("L1TD1")
resnp <- resnp[which(abs(resnp$padj) > 0),]
# include selectLab = selected for lable L1TD1 or not
vol <- EnhancedVolcano(resnp, x = "log2FoldChange", y = "padj", lab = rownames(resnp), title = 'Volcano Plot of Normal Colon Epithelium Tissue vs Adenoma Tissue', drawConnectors = TRUE, widthConnectors = 0.75)
vol
respa <- respa[which(abs(respa$padj) > 0),]
vol2 <- EnhancedVolcano(respa, x = "log2FoldChange", y = "padj", lab = rownames(respa), title = 'Volcano Plot of Primary CRC Tissue vs Adenoma Tissue', drawConnectors = TRUE, widthConnectors = 0.75)
vol2
```

![paired_normal_vs_adenoa_vol1](https://github.com/yerimeeei/Bioinformatics/assets/134043926/3ebd58c0-3405-454e-900c-30b0dac975a2)

![paired_primary_vs_adenoa_vol1](https://github.com/yerimeeei/Bioinformatics/assets/134043926/cb76dc42-6c97-4995-aa13-0c506a905932)

### Venn Diagram

```r
# Venn Diagram
## Data preparation
pval_threshold <- 0.05
na <- setdiff(row.names(resnp[which(resnp$padj <= pval_threshold), ]), row.names(resnt[which(resnt$padj <= pval_threshold), ]))
np <- setdiff(row.names(resnt[which(resnt$padj <= pval_threshold), ]), row.names(resnp[which(resnp$padj <= pval_threshold), ]))
common <- intersect(row.names(resnp[which(resnp$padj <= pval_threshold), ]), row.names(resnt[which(resnt$padj <= pval_threshold), ]))

## Venn-diagram using the `VennDiagram` library
library(VennDiagram)
help("venn.diagram")
venn <- venn.diagram(x = list(na = row.names(resnp[which(resnp$padj <= pval_threshold), ]), 
                              np = row.names(resnt[which(resnt$padj <= pval_threshold), ])), 
                     filename = "venn.png",
                     category.names = c("normal vs adenoma", "normal vs primary"), 
                     fill = c("cornflowerblue", "green"),
                     alpha = c(0.5, 0.5),
                     cex = 1.5,
                     fontfamily = "sans",
                     fontface = "bold",
                     cat.fontfamily = "sans",
                     cat.fontface = "bold",
                     cat.cex = 1.5,
                     margin = 0.05,
                     main = "Overlap of DEGs",
                     main.cex = 1.5,
                     main.fontfamily = "sans",
                     main.fontface = "bold",
                     euler.d = TRUE,
                     counts = list(
                       c(length(na)),
                       c(length(np)),
                       c(length(common))
                     ),
                     scaled = TRUE)
```

![Untitled](https://github.com/yerimeeei/Bioinformatics/assets/134043926/f98dc0b8-f73e-4691-bdca-6cd35be05a82)

## Motif Analysis

### POSSUM (Positional Weight Matrix Suite)

- used to predict the binding affinity of transcription factors.
- POSSUM uses a position weight matrix (PWM) to predict the binding affinity of a transcription factor to a DNA sequence. The PWM is a mathematical representation of the binding preferences of a transcription factor at each position in a DNA sequence.
- The output of POSSUM includes a score that represents the predicted binding affinity, as well as other information about the predicted binding site.

```bash
/storage2/temp/labs/jwlab/LUNG/JASPAR2022_CORE_clear.txt
/storage2/temp/labs/jwlab/LUNG/possum-src/possum
```

<img width="385" alt="Screenshot_2023-07-03_at_10 31 55_AM" src="https://github.com/yerimeeei/Bioinformatics/assets/134043926/329644c2-4ae6-4ece-93eb-3b5b769a03e6">

![Untitled 1](https://github.com/yerimeeei/Bioinformatics/assets/134043926/389ebb58-3773-4a04-82c7-48620f93c421)

- cut-off point?!

## Analysis

- Get potential gene list that only expressed high in normal vs adenoma but not in normal vs primary.
    
  <img width="644" alt="Screenshot_2023-07-03_at_1 34 54_PM" src="https://github.com/yerimeeei/Bioinformatics/assets/134043926/4701b864-1457-4f9b-8d48-5f3c74ea2478">


```r
na <- setdiff(row.names(resnp[which(resnp$padj < 0.00001 & resnp$log2FoldChange < -2), ]), row.names(resnt[which(resnt$padj < 0.00001 & resnt$log2FoldChange < -2), ]))
grep("L1TD1", na, value = TRUE)
np <- setdiff(row.names(resnt[which(resnt$padj < 0.00001 & resnt$log2FoldChange < -2), ]), row.names(resnp[which(resnp$padj < 0.00001 & resnp$log2FoldChange < -2), ]))

"L1TD1"    "FGFRL1"    "ART3"    "ENC1"      "MGC32805"  "SLC12A2"   "MSX2"     
"MET"      "NEBL"      "APIP"    "TCN1"      "SLC22A11"  "SPTBN2"    "SMAD9"    
"NETO2"    "RNF43.1"   "AXIN2"   "LINC00668" "SCML2P1"   "ZNRF3"
```

| L1TD1 | 1 | LINE-1 type transposase domain containing 1 | protein-coding gene |  |
| --- | --- | --- | --- | --- |
| FGFRL1 | 2 | Fibroblast growth factor receptor-like 1 | protein-coding gene | member of the fibroblast growth factor receptor (FGFR) family, where amino acid sequence is highly conserved between members and throughout evolution |
| ART3 | 3 | ADP-ribosyltransferase 3 (inactive) | protein | encoded protein catalyzes a reversible reaction which modifies proteins by the addition or removal of ADP-ribose to an arginine residue to regulate the function of the modified protein |
| ENC1 | 4 | Ectodermal-Neural Cortex 1 | protein-coding gene | encoded protein plays a role in the oxidative stress response as a regulator of the transcription factor Nrf2, and expression of this gene may play a role in malignant transformation |
| MGC32805 | 5 | Uncharacterized LOC153163 | RNA Gene | affiliated with the lncRNA class |
| SLC12A2 | 6 | Solute Carrier Family 12 Member 2 | protein-coding gene | protein encoded by this gene mediates sodium and chloride transport and reabsorption |
| MSX2 | 7 | Msh Homeobox 2 | protein-coding gene | gene encodes a member of the muscle segment homeobox gene family. The encoded protein is a transcriptional repressor whose normal activity may establish a balance between survival and apoptosis of neural crest-derived cells required for proper craniofacial morphogenesis |
| MET | 8 | MET Proto-Oncogene, Receptor Tyrosine Kinase | protein-coding gene | receptor tyrosine kinase |
| NEBL | 9 | Nebulette | protein-coding gene | encodes a nebulin like protein that is abundantly expressed in cardiac muscle |
| APIP | 10 | APAF1 Interacting Protein | protein-coding gene | Enables identical protein binding activity; methylthioribulose 1-phosphate dehydratase activity; and zinc ion binding activity |
| TCN1 | 11 | Transcobalamin 1 | protein-coding gene | encodes a member of the vitamin B12-binding protein family |
| SLC22A11 | 12 | Solute Carrier Family 22 Member 11 | protein-coding gene | sodium-independent transport and excretion of organic anions, some of which are potentially toxic |
| SPTBN2 | 13 | Spectrin Beta, Non-Erythrocytic 2 | protein-coding gene | regulates the glutamate signaling pathway by stabilizing the glutamate transporter EAAT4 at the surface of the plasma membrane |
| SMAD9 | 14 | SMAD Family Member 9 | protein-coding gene | MAD family, which transduces signals from TGF-beta family members |
| NETO2 | 15 | Neuropilin And Tolloid Like 2 | protein-coding gene | predicted transmembrane protein containing two extracellular CUB domains followed by a low-density lipoprotein class A (LDLa) domain |
| https://www.ncbi.nlm.nih.gov/gene/54894 | 16 | Ring Finger Protein 43 | protein-coding gene | protein encoded by this gene is thought to negatively regulate Wnt signaling, and expression of this gene results in an increase in ubiquitination of frizzled receptors, an alteration in their subcellular distribution, resulting in reduced surface levels of these receptors. |
| AXIN2 | 17 | Axin 2 | protein-coding gene | regulation of the stability of beta-catenin in the Wnt signaling pathway, like its rodent homologs, mouse conductin/rat axil |
| LINC00668 | 18 | Long Intergenic Non-Protein Coding RNA 668 | RNA Gene | affiliated with the lncRNA class |
| SCML2P1 | 19 | SCML2 Pseudogene 1 | Pseudogene |  |
| ZNRF3 | 20 | Zinc And Ring Finger 3 | protein-coding gene | related pathways are https://pathcards.genecards.org/card/signaling_by_wnt and https://pathcards.genecards.org/card/signal_transduction |
- L1TD1 TF binding affinity desencing order

| name | class | family |  |
| --- | --- | --- | --- |
| ETV5::FOXI1 | Tryptophan cluster factors::Fork head/winged helix factors | Ets-related::FOX | 10.8 |
| HOXC11 | Homeo domain factors | HOX | 10.1 |
| HMBOX1 | Homeo domain factors | POU domain factors | 9.42 |
| IKZF1 | C2H2 zinc finger factors | Factors with multiple dispersed zinc fingers | 9.32 |
| HOXC9 | Homeo domain factors | HOX | 9.27 |
| POU2F1 | Homeo domain factors | POU domain factors | 8.89 |
| ZNF331 | C2H2 zinc finger factors | More than 3 adjacent zinc fingers | 8.80 |
| ERF::FOXO1 | Tryptophan cluster factors::Fork head/winged helix factors | Ets-related::FOX | 8.65 |
| GCM2 | GCM domain factors  | GCM | 8.48 |
| NR2F1 | Nuclear receptors with C4 zinc fingers  | RXR-related receptors (NR2)  | 8.44 |
- Potential TF list that is related to the lists that are only highly expressed in adenoma.

### Match motif (gene list only highly expressed in adenoma and L1TD1 promoter/gene)

- [x]  First step is to work out which on differentially expressed gene is a transcript factor. You can check against this database ([http://humantfs.ccbr.utoronto.ca/](http://humantfs.ccbr.utoronto.ca/)).
- [x]  Next you see if you can match their motif to what you find in L1TD1 promoter/gene.

| http://humantfs.ccbr.utoronto.ca/download/v_1.01/Readme_TFs_v_1.01.txt | DBD Family | Is TF? | TF Assessment | Binding Activity | Motif Status | Notes | Comments |
| --- | --- | --- | --- | --- | --- | --- | --- |
| MSX2 | Homeodomain | Yes | Known motif | 1 Monomer or homomultimer | High-throughput in vitro |  |  |
| MET | Unknown | No | Unlikely to be sequence specific TF | 4 Not a DNA binding protein | No motif |  | Its a cell surface receptor and kinase - included only because Vasquerizas 2009 included it with an x. |
| SMAD9 | SMAD | Yes | Known motif | 2 Obligate heteromer | In vivo/Misc source | Only known motifs are from Transfac or HocoMoco - origin is uncertain |  |

```r
"L1TD1"    "FGFRL1"    "ART3"    "ENC1"      "MGC32805"  "SLC12A2"   "MSX2"     
"MET"      "NEBL"      "APIP"    "TCN1"      "SLC22A11"  "SPTBN2"    "SMAD9"    
"NETO2"    "RNF43.1"   "AXIN2"   "LINC00668" "SCML2P1"   "ZNRF3"
```

```r
"MGC32805"  "LINC00668" "SCML2P1" 
```

#
