# Day 13 &14 L1TD1 gene expression Normal vs Adenoma Motif Analysis

### To Do

- [ ]  L1TD1 gene → for further studies, need to choose whether I want to do for other lines or only L1 → Study and **Read literature of L1 cancer basic**
- [x]  Find relevant data [SRA Explorer](https://sra-explorer.info/#)
- [x]  Worth doing PCA plot
- [ ]  Look at DE list Adenoma vs. Normal for TFs (up-regulation)
- [ ]  Work out motifs of up regulated TFs (JASPAR/ Trcasfae)
- [ ]  Search for motifs in L1TD1 promoter (TSS +/- 0.5kb)

### SRA Data

- [Colorectal cancer cell line](https://www.ncbi.nlm.nih.gov/Traces/study/?acc=SRP055442&o=acc_s%3Aa)
- [Colon cell CAPG Knockdown vs WT](https://www.ncbi.nlm.nih.gov/Traces/study/?acc=SRP441703&o=acc_s%3Aa)
- [Colon Adenoma organoid cell](https://www.ncbi.nlm.nih.gov/Traces/study/?acc=ERP015684&o=acc_s%3Aa)
- [**Gene expression analysis of human adenomas.**](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE72820)
    - **[Genomic and mutational profiling of human colon adenomas reveals early driver mutations and a TGF-β-CEA regulated profile](https://aacrjournals.org/cancerres/article/75/15_Supplement/67/604880/Abstract-67-Genomic-and-mutational-profiling-of)**
        - **Transcriptome sequence analysis was performed in 7 pairs of control-test matched adenoma (6 TVA and 1 SSA)**
        - Sessile serrated adenomas (SSAs) and tubulovillous adenomas (TVAs) are types of polyps that can develop in the colon or rectum.
        - SSAs are a type of premalignant polyp that has a distinct microscopic appearance and is characterized by serrated crypts on the surface of the polyp. They are typically flat or sessile (i.e., they do not have a stalk) and can be difficult to detect during colonoscopy. SSAs have been associated with an increased risk of developing colorectal cancer, particularly in individuals over the age of 50.
        - TVAs are another type of premalignant polyp that can be found in the colon or rectum. They have a glandular structure and can be either sessile or pedunculated (i.e., they have a stalk). TVAs are more common than SSAs and are also associated with an increased risk of developing colorectal cancer.
    - ****[Mutational Profiles Reveal an Aberrant TGF-β-CEA Regulated Pathway in Colon Adenomas](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0153933)****
- [Transcriptome profile of human colorectal adenomas by array](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE8671)
- [Human Transcriptome Array 2.0 (HTA) from healthy colonic, colorectal adenoma and colorectal cancer tissue](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE100179)
- [adjacent normal colon vs normal colon](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE95132)

### PCA Plot

![L1TD1_PCA_plot](https://github.com/yerimeeei/Bioinformatics/assets/134043926/af15ba0c-e9d7-47e8-bfec-e752558873dd)

## Gene expression analysis of human adenomas

- [GEO](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE72820)

### [SRA Run](https://www.ncbi.nlm.nih.gov/Traces/study/?acc=SRP063499&o=acc_s%3Aa)

```r
awk '{print"/storage2/jwlab/sandy/sratoolkit.3.0.1-ubuntu64/bin/fastq-dump --split-3 --gzip "$0}' /storage2/jwlab/yerim/adenomas/SRR_Acc_List.txt > srr_download.sh
parallel -j 2 < srr_download.sh

vi run_STAR.sh
fq1=$1
fq2=$2
name=$3

mkdir /storage2/jwlab/yerim/adenomas/BAM/$name
~/storage2/jwlab/yerim/STAR --runThreadN 5 --genomeDir /storage2/jwlab/yerim/GenomeDir/genome_index/ --readFilesIn $fq1 $fq2 --readFilesCommand zcat --outFileNamePrefix /storage2/jwlab/yerim/adenomas/BAM/$name/ --outSAMtype BAM SortedByCoordinate --outSAMattributes All --outSAMattrIHstart 0 --outFilterMultimapNmax 100 --outFilterScoreMinOverLread 0.4 --outFilterMatchNminOverLread 0.4 --clip3pNbases 0 --winAnchorMultimapNmax 100 --alignEndsType EndToEnd --alignEndsProtrude 100 DiscordantPair --chimSegmentMin 250 --sjdbOverhang 99 --twopassMode Basic

awk '{print"bash run_STAR.sh /storage2/jwlab/yerim/adenomas/fastq/"$0"_1.fastq.gz /storage2/jwlab/yerim/adenomas/fastq/"$0"_2.fastq.gz "$0}' SRR_Acc_List.txt > fastq_list.sh
parallel -j 2 < fastq_list.sh

vi fc.sh
name=$1
bam=$2

mkdir /storage2/jwlab/yerim/adenomas/fc_output/$name
/storage2/jwlab/yerim/featureCounts -T 5 -p --countReadPairs -O -a /storage2/jwlab/yerim/gencode.L1.gtf -o /storage2/jwlab/yerim/adenomas/fc_output/$name/fc_result.txt $bam

awk '{print"/storage2/jwlab/yerim/adenomas/BAM/"$0"/Aligned.sortedByCoord.out.bam"}' /storage2/jwlab/yerim/adenomas/SRR_Acc_List.txt > bam_list.sh
awk '{print"bash fc.sh "$0" /storage2/jwlab/yerim/adenomas/BAM/"$0"/Aligned.sortedByCoord.out.bam"}' SRR_Acc_List.txt > bam_fc_list.sh
parallel -j 2 < bam_fc_list.sh
```

### Normalization and Violin plot

```r
fc_result_list <- read.table("/storage2/jwlab/yerim/adenomas/fc_output/fc_result_list", quote="\"", comment.char="")
fc_result_list$name <- unlist(sapply(strsplit(fc_result_list$V1, "/", fixed=TRUE), function(x) x[6], simplify=FALSE))
SraRunTable <- read.csv ("/storage2/jwlab/yerim/adenomas/fc_output/SraRunTable.txt" )
fc_result_list$source <- unlist(SraRunTable[match(fc_result_list$name,SraRunTable$Run),"Tissue"])
fc_result_list <- fc_result_list[,-1]
rownames(fc_result_list) <- fc_result_list[,1]

SRR18_fc_result <- read.delim("/storage2/jwlab/yerim/adenomas/fc_output/SRR2316018/fc_result.txt", comment.char="#")
SRR19_fc_result <- read.delim("/storage2/jwlab/yerim/adenomas/fc_output/SRR2316019/fc_result.txt", comment.char="#")
SRR20_fc_result <- read.delim("/storage2/jwlab/yerim/adenomas/fc_output/SRR2316020/fc_result.txt", comment.char="#")
SRR21_fc_result <- read.delim("/storage2/jwlab/yerim/adenomas/fc_output/SRR2316021/fc_result.txt", comment.char="#")
SRR22_fc_result <- read.delim("/storage2/jwlab/yerim/adenomas/fc_output/SRR2316022/fc_result.txt", comment.char="#")
SRR23_fc_result <- read.delim("/storage2/jwlab/yerim/adenomas/fc_output/SRR2316023/fc_result.txt", comment.char="#")
SRR24_fc_result <- read.delim("/storage2/jwlab/yerim/adenomas/fc_output/SRR2316024/fc_result.txt", comment.char="#")
SRR25_fc_result <- read.delim("/storage2/jwlab/yerim/adenomas/fc_output/SRR2316025/fc_result.txt", comment.char="#")
SRR26_fc_result <- read.delim("/storage2/jwlab/yerim/adenomas/fc_output/SRR2316026/fc_result.txt", comment.char="#")
SRR27_fc_result <- read.delim("/storage2/jwlab/yerim/adenomas/fc_output/SRR2316027/fc_result.txt", comment.char="#")
SRR28_fc_result <- read.delim("/storage2/jwlab/yerim/adenomas/fc_output/SRR2316028/fc_result.txt", comment.char="#")
SRR29_fc_result <- read.delim("/storage2/jwlab/yerim/adenomas/fc_output/SRR2316029/fc_result.txt", comment.char="#")
SRR30_fc_result <- read.delim("/storage2/jwlab/yerim/adenomas/fc_output/SRR2316030/fc_result.txt", comment.char="#")
SRR31_fc_result <- read.delim("/storage2/jwlab/yerim/adenomas/fc_output/SRR2316031/fc_result.txt", comment.char="#")
result <- SRR18_fc_result[,c(1,7)]
colnames(result)[2] <- "SRR2316018"
result <- cbind(result, SRR19_fc_result$X.storage2.jwlab.yerim.adenomas.BAM.SRR2316019.Aligned.sortedByCoord.out.bam)
colnames(result)[3] <- "SRR2316019"
result <- cbind(result, SRR20_fc_result$X.storage2.jwlab.yerim.adenomas.BAM.SRR2316020.Aligned.sortedByCoord.out.bam)
colnames(result)[4] <- "SRR2316020"
result <- cbind(result, SRR21_fc_result$X.storage2.jwlab.yerim.adenomas.BAM.SRR2316021.Aligned.sortedByCoord.out.bam)
colnames(result)[5] <- "SRR2316021"
result <- cbind(result, SRR22_fc_result$X.storage2.jwlab.yerim.adenomas.BAM.SRR2316022.Aligned.sortedByCoord.out.bam)
colnames(result)[6] <- "SRR2316022"
result <- cbind(result, SRR23_fc_result$X.storage2.jwlab.yerim.adenomas.BAM.SRR2316023.Aligned.sortedByCoord.out.bam)
colnames(result)[7] <- "SRR2316023"
result <- cbind(result, SRR24_fc_result$X.storage2.jwlab.yerim.adenomas.BAM.SRR2316024.Aligned.sortedByCoord.out.bam)
colnames(result)[8] <- "SRR2316024"
result <- cbind(result, SRR25_fc_result$X.storage2.jwlab.yerim.adenomas.BAM.SRR2316025.Aligned.sortedByCoord.out.bam)
colnames(result)[9] <- "SRR2316025"
result <- cbind(result, SRR26_fc_result$X.storage2.jwlab.yerim.adenomas.BAM.SRR2316026.Aligned.sortedByCoord.out.bam)
colnames(result)[10] <- "SRR2316026"
result <- cbind(result, SRR27_fc_result$X.storage2.jwlab.yerim.adenomas.BAM.SRR2316027.Aligned.sortedByCoord.out.bam)
colnames(result)[11] <- "SRR2316027"
result <- cbind(result, SRR28_fc_result$X.storage2.jwlab.yerim.adenomas.BAM.SRR2316028.Aligned.sortedByCoord.out.bam)
colnames(result)[12] <- "SRR2316028"
result <- cbind(result, SRR29_fc_result$X.storage2.jwlab.yerim.adenomas.BAM.SRR2316029.Aligned.sortedByCoord.out.bam)
colnames(result)[13] <- "SRR2316029"
result <- cbind(result, SRR30_fc_result$X.storage2.jwlab.yerim.adenomas.BAM.SRR2316030.Aligned.sortedByCoord.out.bam)
colnames(result)[14] <- "SRR2316030"
result <- cbind(result, SRR31_fc_result$X.storage2.jwlab.yerim.adenomas.BAM.SRR2316031.Aligned.sortedByCoord.out.bam)
colnames(result)[15] <- "SRR2316031"
rownames(result) <- result$Geneid
result <- result[,c(-1)]

library(edgeR)
dgList <- calcNormFactors(result, method="TMM")
tmm <- cpm(result,dgList, log=TRUE, prior.count=1)

tmm <- cbind(tmm, SRR18_fc_result$Geneid)
colnames(tmm)[15] <- "Geneid"
grep("ENSG00000240563", row.names(tmm), value = TRUE)
L1TD1 <- tmm[row.names(tmm) == "ENSG00000240563.2"]

L1TD1 <- cbind(L1TD1, fc_result_list$name)
colnames(L1TD1)[2] <- "name"
L1TD1 <- cbind(L1TD1, fc_result_list$source)
colnames(L1TD1)[3] <- "sample"
L1TD1 <- L1TD1[c(-15),]
colnames(L1TD1)[1] <- "value"
df <- as.data.frame(L1TD1)

library(ggplot2)
vplot <- ggplot(df, aes(x=sample, y=value, group=sample, fill=sample)) 
vplot <- vplot + geom_violin(trim=FALSE, fill="gray") 
vplot <- vplot + labs(title="L1TD1 expression in normal mucosa vs adenoma", x="Sample Identities", y = "log2(CPM+1)") 
vplot <- vplot + geom_boxplot(width=0.1)+ theme_classic() + scale_x_discrete(limits=c("Human normal mucosa", "Human colon adenoma"))
vplot
```

### Differential Analysis (DESeq2)

- Volcano plot
- PCA plot

```r
# DESeq2 Differential Analysis
library(DESeq2)
all(colnames(result) %in% rownames(fc_result_list))
fc_result_list <- fc_result_list[c(1,13,14,2,12,10,3,4,5,6,7,8,11,9),]
all(colnames(result) == rownames(fc_result_list))

dds <- DESeqDataSetFromMatrix(countData = result, colData = fc_result_list, design = ~ source)
dds$source <- relevel(dds$source, ref = "Human normal mucosa")
dds <- DESeq(dds)
res <- results(dds)
summary (res)
head(results(dds, tidy=TRUE))
res <- res[order(res$padj),]
head(res)
res <- results(dds, contrast = c("source", "Human normal mucosa", "Human colon adenoma"))
#volcano plot
install.packages("devtools")
devtools::install_github('kevinblighe/EnhancedVolcano')
library(EnhancedVolcano)
selected = c("ENSG00000240563.2")
vol <- EnhancedVolcano(res, x = "log2FoldChange", y = "padj", lab = row.names(res), selectLab = selected, title = 'Volcano Plot of Human Normal Mucosa vs Human Colon Adenoma', xlab = bquote(~Log[2]~ 'fold change'), legendPosition = 'right', legendLabSize = 12, legendIconSize = 4.0, drawConnectors = TRUE, widthConnectors = 0.75)
#PCA plot
vsdata <- vst(dds, blind=FALSE)
plotPCA(vsdata, intgroup="source")
```

<img width="822" alt="Screenshot_2023-06-25_at_4 22 05_PM" src="https://github.com/yerimeeei/Bioinformatics/assets/134043926/a4b95df1-5745-4185-ac7e-6f39ad869169">

- from the result where values of differential analysis are null (NA), remove it for volcano plots.

### Principal Components Analysis (PCA)

### Motif Analysis

- [JASPAR](https://jaspar.genereg.net/search?q=SPI1&collection=CORE&tax_group=vertebrates)
- [Toolkit for Cistrome Data Browser](http://dbtoolkit.cistrome.org)
