# Day 15 & 16 & 17 L1TD1 gene expression Motif Analysis

### To do

- [ ]  L1TD1 gene → for further studies, need to choose whether I want to do for other lines or only L1 → Study and **Read literature of L1 cancer basic**
- [ ]  Look at DE list Adenoma vs. Normal for TFs (up-regulation)
- [ ]  Work out motifs of up regulated TFs (JASPAR/ Trcasfae)
- [x]  Search for motifs in L1TD1 promoter (transcription start site (TSS) +/- 0.5kb)

```r
library(EnhancedVolcano)
selected = c("ENSG00000240563.2")
res <- res[which(abs(res$log2FoldChange) > 0),]
vol <- EnhancedVolcano(res, x = "log2FoldChange", y = "pvalue", lab = row.names(res), selectLab = selected, title = 'Volcano Plot of Human Normal Mucosa vs Human Colon Adenoma', drawConnectors = TRUE, widthConnectors = 0.75)
```

## DE list Adenoma vs. Normal for TFs (up-regulation)

### 7 paired N vs A

```r
result.pca <- result[which(abs(result) >= 1),]
result.pca <- result.pca[rowSums(result.pca[-1] != result.pca[[2]], na.rm = TRUE) != 0,]
result.pca <- result.pca[str_detect(rownames(result.pca),"ENSG"),]
result.pca <- result.pca[,c(-7,-12)]

library(org.Hs.eg.db)
keys <- substr(rownames(result.pca), 1,15)
symbols <- mapIds(org.Hs.eg.db, keys = keys,
                  column = c('SYMBOL'), keytype = 'ENSEMBL')
symbols <- symbols[!is.na(symbols)]
symbols <- symbols[match(substr(rownames(result.pca), 1,15), names(symbols))]
rownames(result.pca) = make.names(symbols, unique = TRUE)
keep <- !is.na(substr(rownames(result.pca), 1,15))
result.pca <- result.pca[keep,]

dds <- DESeqDataSetFromMatrix(countData = result.pca, colData = fc_result_list[c(-7,-12),], design = ~ source)
dds$source <- relevel(dds$source, ref = "Human normal mucosa")
dds <- DESeq(dds)
res <- results(dds)
head(results(dds, tidy=TRUE))
res <- res[order(res$padj),]
res <- results(dds, contrast = c("source", "Human normal mucosa", "Human colon adenoma"))

library(EnhancedVolcano)
selected = c("L1TD1")
res <- res[which(abs(res$padj) > 0),]
vol <- EnhancedVolcano(res, x = "log2FoldChange", y = "pvalue", lab = rownames(res), title = 'Volcano Plot of Human Normal Mucosa vs Human Colon Adenoma', drawConnectors = TRUE, widthConnectors = 0.75)
vol
```

![35c0b670-9220-4388-896b-a5d9c17b81bf](https://github.com/yerimeeei/Bioinformatics/assets/134043926/bbb229fc-c4a5-434a-a381-747106ebb533)

### normal colon epithelium vs adenoma (P vs N vs A)

![f0cc5938-2d11-49c7-8bca-1d5dbd960abb](https://github.com/yerimeeei/Bioinformatics/assets/134043926/7fbcbbc4-55fd-40a1-91b4-725873849c59)

![b3d54ba5-e8be-42a6-a875-d0cd132d653c](https://github.com/yerimeeei/Bioinformatics/assets/134043926/f07538e4-1ae5-47a5-8558-5d199ef97aaf)

FOXQ1

![chr6_1312097_1314757_NM_033260_FOXQ1hg381k_factor](https://github.com/yerimeeei/Bioinformatics/assets/134043926/e0282072-f9a9-44bd-9824-1b76628bd650)
.png)

| name | class | family |  |
| --- | --- | --- | --- |
| IRF8 | Tryptophan cluster factors | Interferon-regulatory factors | 11.3 |
| OSR1 | C2H2 zinc finger factors | Other factors with up to three adjacent zinc fingers | 11.0 |
| ZNF93 | C2H2 zinc finger factors | More than 3 adjacent zinc fingers | 10.8 |
| LHX2 | Homeo domain factors | HD-LIM | 10.8 |
| TFAP2A | Basic helix-span-helix factors (bHSH) | AP-2 | 10.2 |
| DBP | Basic leucine zipper factors (bZIP) | CEBP-related | 9.67 |
| ZNF93 | C2H2 zinc finger factors | More than 3 adjacent zinc fingers | 9.65 |
| FOXE1 | Fork head/winged helix factors | FOX | 9.63 |
| ZBTB12 | C2H2 zinc finger factors | More than 3 adjacent zinc fingers | 9.55 |

MMP7

![chr11_102520507_102530746_NM_002423_MMP7hg3810k_factor](https://github.com/yerimeeei/Bioinformatics/assets/134043926/cbca00d6-09a0-4bef-8dd7-de0f83a798aa)

| name | class | family |  |
| --- | --- | --- | --- |
| TFAP2B | Basic helix-span-helix factors (bHSH) | AP-2 | 9.91 |
| TFAP2B | Basic helix-span-helix factors (bHSH) | AP-2 | 9.80 |
| TFAP2C | Basic helix-span-helix factors (bHSH) | AP-2 | 9.51 |
| IRF6 | Tryptophan cluster factors | Interferon-regulatory factors | 9.47 |
| TFAP2C | Basic helix-span-helix factors (bHSH) | AP-2 | 9.36 |
| ATF2 | Basic leucine zipper factors (bZIP) | Jun-related | 9.11 |

## Principal Components Analysis (PCA)

```r
result_nec <- result[str_detect(rownames(result),"ENSG"),]
dds <- DESeqDataSetFromMatrix(countData = result_nec, colData = fc_result_list, design = ~ source)
dds$source <- relevel(dds$source, ref = "Human normal mucosa")
dds <- DESeq(dds)
res <- results(dds)
summary(res)
head(results(dds, tidy=TRUE))
res <- res[order(res$padj),]
DEGs <- res[which(res$padj < 0.05 & abs(res$log2FoldChange) > 1),]

#PCA plot
library(factoextra)
result.pca <- result[which(abs(result) >= 1),]
result.pca <- result.pca[rowSums(result.pca[-1] != result.pca[[2]], na.rm = TRUE) != 0,]
result.pca <- result.pca[str_detect(rownames(result.pca),"ENSG"),]
pca <- prcomp(t(result.pca), scale=TRUE)
attributes(pca)
pca$center
pca$scale
print(pca)
library(ggbiplot)
g <- ggbiplot(pca,
              obs.scale = 1,
              var.scale = 1,
              groups = as.factor(fc_result_list$source),
              ellipse = TRUE,
              circle = TRUE,
              ellipse.prob = 0.68, varname.size = 0) #, var.axes=FALSE
g <- g + scale_color_discrete(name = '')
g <- g + theme(legend.direction = 'horizontal',
               legend.position = 'top')
print(g)

library(PCAtools)
all(colnames(result) == rownames(fc_result_list))
p <- pca(result, metadata = fc_result_list, removeVar = 0.1)
biplot(p, showLoadings = TRUE,
       labSize = 5, pointSize = 5, sizeLoadingsNames = 5)
```
![123ca58a-40ae-4afc-b79d-2259b785619f](https://github.com/yerimeeei/Bioinformatics/assets/134043926/aed886ec-1259-4af1-8fa6-cb81a4de06ee)

## Motif Analysis

```bash
/storage2/temp/labs/jwlab/LUNG/JASPAR2022_CORE_clear.txt
/storage2/temp/labs/jwlab/LUNG/possum-src/possum
```

- [JASPAR](https://jaspar.genereg.net/search?q=SPI1&collection=CORE&tax_group=vertebrates)
- [Cistrome](http://cistrome.org/db/#/)
- [Toolkit for Cistrome Data Browser](http://dbtoolkit.cistrome.org)

### motifs in L1TD1 promoter (all)

```bash
/storage2/temp/labs/jwlab/LUNG/possum-src/possum /storage2/temp/labs/jwlab/LUNG/JASPAR2022_CORE_clear.txt /storage2/jwlab/yerim/L1TD1.fa -t 5 -o L1TD1_output.txt
awk '{$1=$3=$4=$5=$6=$7=""; print $0}' L1TD1_output.txt > extract_L1TD1.txt
sort -k2n extract_L1TD1.txt > sort_L1TD1.txt
```

| name | class | family |  |
| --- | --- | --- | --- |
| ZIC5 | C2H2 zinc finger factors | More than 3 adjacent zinc fingers | 14.1 |
| PRDM9 | C2H2 zinc finger factors | Factors with multiple dispersed zinc fingers | 13.2 |
| ZKSCAN3 | C2H2 zinc finger factors | More than 3 adjacent zinc fingers | 12.7 |
| ETV5::FIGLA | Tryptophan cluster factors::Basic helix-loop-helix factors (bHLH) | Ets-related::Tal-related | 12.0 |
| DMRTA1 | DM-type intertwined zinc finger factors | DMRT factors | 12.0 |
| IKZF1 | C2H2 zinc finger factors | Factors with multiple dispersed zinc fingers | 11.9 |
| ERF::FIGLA | Tryptophan cluster factors::Basic helix-loop-helix factors (bHLH) | Ets-related::Tal-related | 11.9 |
| POU1F1 | Homeo domain factors | POU domain factors | 11.7 |
| CUX1 | Homeo domain factors | HD-CUT | 11.7 |
| ZNF669 | C2H2 zinc finger factors | More than 3 adjacent zinc fingers | 11.6 |

### motifs in L1TD1 promoter (exons)

| name | class | family |  |
| --- | --- | --- | --- |
| ZKSCAN3 | C2H2 zinc finger factors | More than 3 adjacent zinc fingers | 12.7 |
| Zic1::Zic2 | C2H2 zinc finger factors | More than 3 adjacent zinc fingers::More than 3 adjacent zinc fingers | 11.3 |
| ZNF136 | C2H2 zinc finger factors | More than 3 adjacent zinc fingers | 10.9 |
| HOXD12::ELK1 | Homeo domain factors::Tryptophan cluster factors | HOX::Ets-related | 10.9 |
| ZFP42 | C2H2 zinc finger factors | More than 3 adjacent zinc fingers | 10.8 |
| TEAD3 | TEA domain factors | TEF-1-related factors | 10.8 |
| TBX19 | T-Box factors | Brachyury-related factors | 10.8 |
| ZBTB18 | C2H2 zinc finger factors | More than 3 adjacent zinc fingers | 10.6 |
| TBX19 | T-Box factors | Brachyury-related factors | 10.6 |
| PAX6 | Paired box factors | Paired plus homeo domain | 10.6 |

### motifs in L1TD1 promoter (exon 1)

| name | class | family |  |
| --- | --- | --- | --- |
| HOXC11 | Homeo domain factors | HOX | 10.5 |
| HMBOX1 | Homeo domain factors | POU domain factors | 9.10 |
| ZBTB32 | C2H2 zinc finger factors | Other factors with up to three adjacent zinc fingers | 8.71 |
| KLF7 | C2H2 zinc finger factors | Three-zinc finger Kruppel-related | 8.61 |
| MEIS3 | Homeo domain factors | TALE-type homeo domain factors | 8.46 |
| MEIS2 | Homeo domain factors | TALE-type homeo domain factors | 8.21 |
| FOXL1 | Fork head/winged helix factors | FOX | 7.75 |
| FOXI1 | Fork head/winged helix factors | FOX | 7.64 |
| E2F8 | Fork head/winged helix factors | E2F | 7.24 |
| FOX06 | Fork head/winged helix factors | FOX | 7.11 |

### motifs in L1TD1 promoter (TSS +/-)

| name | class | family |  |
| --- | --- | --- | --- |
| ETV5::FOXI1 | Tryptophan cluster factors::Fork head/winged helix factors | Ets-related::FOX | 10.8 |
| HOXC11 | Homeo domain factors | HOX | 10.1 |
| HMBOX1 | Homeo domain factors | POU domain factors | 9.42 |
| IKZF1 | C2H2 zinc finger factors | Factors with multiple dispersed zinc fingers | 9.32 |
| HOXC9 | Homeo domain factors | HOX | 9.27 |
| POU2F1 | Homeo domain factors | POU domain factors | 8.89 |
| ZNF331 | C2H2 zinc finger factors | More than 3 adjacent zinc fingers | 8.80 |
| ERF::FOXO1 | Tryptophan cluster factors::Fork head/winged helix factors | Ets-related::FOX | 8.65 |
| GCM2 | GCM domain factors  | GCM | 8.48 |
| NR2F1 | Nuclear receptors with C4 zinc fingers  | RXR-related receptors (NR2)  | 8.44 |

## 0628 meeting TODO

- [ ]  **paired sample (N vs A) differential analysis**
- [ ]  **venn diagram for (N vs A) & (A vs CA)**
- [x]  **motif analysis of upstream +/-**

```bash
>hg38_dna range=chr1:62194583-62194919 5'pad=0 3'pad=0 strand=+ repeatMasking=none
TTGGGGTTCAGGCCTGGGTGATGAGGCGCTTTAGTGGTGCCTTTATTCAG
ATTTGAGTCTGGAGTCGGGTCTGAGGGGTGGGGCTCGTTCAAGCGGAGGC
ACGGTGGAGACAGAGGGAATTGGACTCAGTCGGGGGGCAGCAATGGAAGT
CAGCAATTGCTTAGGGCGGAATGCGATTTCCGGGGAGGAGGCGCGGGTAT
GTAGACAGAGGGGGTTGGGACACACCAGGAGGGGAGGAGCCAGCCCCAGA
GATCGGGAATCCTCTCAGTCCTTAGTTACAAGGCTCCATCCTCACTTTGT
TCGCTCCTCAGTCGTCCAGGCGGATTCCTTTTTCGCC
```

## Single-cell Analysis

- [read paper](https://www.nature.com/articles/s41588-022-01088-x)
- snRNA data download
- SEURAT R package tool that does for single cell analyses
